### DISCUSSION TITLE
Issues running on Windows Subsytem for Linux (WSL)

### DISCUSSION THREAD

**Discussion by nbgraham** [Kong/insomnia#6180] [Help]
I'm trying to develop on WSL. I was able to bootstrap and run tests successfully, but seeing errors on `npm run app-start`

This is probably not worth investigating, but I wanted to at least raise it, mostly so other people don't try this inside WSL.

# libnss3 error

`libnss3.so: cannot open shared object file: No such file or directory`

Solution:

```
apt install libnss3-dev libgdk-pixbuf2.0-dev libgtk-3-dev libxss-dev
```
https://stackoverflow.com/a/71756304

The [recommendation in the README](https://github.com/nbgraham/insomnia/tree/develop#linux) may have also worked.

# /var/run/dbus/system_bus_socket error

 `Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket`

Solution:
```
$ sudo mkdir -p /run/dbus
$ sudo dbus-daemon --system
```
https://github.com/microsoft/WSL/issues/2941

# GLIBC_2.32 error

```
Error: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found
```

Looks complicated to solve: https://askubuntu.com/a/1228256

**Comment by filfreire** [comment]
@nbgraham I tried WSL a while back and the GLIBC_2.32 specific error in my case happened due to using an "older" distro, like using Ubuntu 20 instead of 22

After upgrading to Ubuntu 22, the GLIBC_2.32 error should be gone.

As for running the `app-start` I'm using [VcsSrvX](https://sourceforge.net/projects/vcxsrv/), following steps in this post: https://www.beekeeperstudio.io/blog/building-electron-windows-ubuntu-wsl2

**Reply by filfreire** [reply]
Update: seems to be working for me for Ubuntu 22 on WSL, example:

![Screenshot 2023-07-28 163324](https://github.com/Kong/insomnia/assets/11976836/bd56ecc7-7f71-49ef-b716-0526525cf797)
